Dapper/DynamicParameters.cs

```diff
@@ -463,11 +463,13 @@ public DynamicParameters Output<T>(T target, Expression<Func<T, object>> express
 
                 if (parameters.TryGetValue(dynamicParamName, out parameter))
                 {
-                    parameter.ParameterDirection = parameter.AttachedParam.Direction = ParameterDirection.InputOutput;
+                    parameter.ParameterDirection = ParameterDirection.InputOutput;
+                    parameter.Size = sizeToSet;
 
-                    if (parameter.AttachedParam.Size == 0)
+                    if (parameter.AttachedParam != null)
                     {
-                        parameter.Size = parameter.AttachedParam.Size = sizeToSet;
+                        parameter.AttachedParam.Direction = ParameterDirection.InputOutput;
+                        parameter.AttachedParam.Size = parameter.AttachedParam.Size == 0 ? sizeToSet : parameter.AttachedParam.Size;

```

Comment:
Agree, only left it the way it was to keep coding style / minimise changes as not sure on the policy for these things.

I'll update the code when I'm more awake tomorrow.